src/components/application_manager/test/request_tracker_test.cc

```diff
@@ -0,0 +1,250 @@
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "utils/macro.h"
+#include "application_manager/request_tracker.h"
+#include "application_manager/mock_request_controller_settings.h"
+#include "interfaces/MOBILE_API.h"
+#include "utils/lock.h"
+#include "utils/conditional_variable.h"
+
+namespace test {
+namespace components {
+namespace request_controller_test {
+
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+class RequestTrackerTestClass : public ::testing::Test {
+ public:
+  RequestTrackerTestClass() : tracker_(mock_request_controller_settings_) {}
+
+  void SetDefaultConstraints() {
+    EXPECT_CALL(mock_request_controller_settings_,
+                app_hmi_level_none_time_scale())
+        .WillRepeatedly(ReturnRef(kDefaultAppHmiLevelNoneRequestsTimeScale));
+
+    EXPECT_CALL(mock_request_controller_settings_,
+                app_hmi_level_none_time_scale_max_requests())
+        .WillRepeatedly(ReturnRef(kDefaultAppHmiLevelNoneTimeScaleMaxRequests));
+
+    EXPECT_CALL(mock_request_controller_settings_, app_time_scale())
+        .WillRepeatedly(ReturnRef(kDefaultAppRequestsTimeScale));
+
+    EXPECT_CALL(mock_request_controller_settings_,
+                app_time_scale_max_requests())
+        .WillRepeatedly(ReturnRef(kDefaultAppTimeScaleMaxRequests));
+  }
+
+ protected:
+  application_manager_test::MockRequestControlerSettings
+      mock_request_controller_settings_;
+
+  application_manager::request_controller::RequestTracker tracker_;
+
+  const uint32_t kDefaultAppHmiLevelNoneRequestsTimeScale = 10;
+  const uint32_t kDefaultAppHmiLevelNoneTimeScaleMaxRequests = 100u;
+  const uint32_t kDefaultAppTimeScaleMaxRequests = 5;
+  const uint32_t kDefaultAppRequestsTimeScale = 200u;
+};
+
+TEST_F(RequestTrackerTestClass, TrackAppRequestInNone_ExpectSuccessTillLimit) {
+  const uint32_t app_id = 1;
+  const mobile_apis::HMILevel::eType none_level =
+      mobile_apis::HMILevel::HMI_NONE;
+
+  SetDefaultConstraints();
+
+  for (size_t i = 0; i < kDefaultAppHmiLevelNoneTimeScaleMaxRequests; ++i) {
```

Comment:
@dev-gh It's wrong to compare different types. Even if they are both unsigned. size_t can be 64bit value and uint32_t is always 32bit.